rllib/algorithms/algorithm.py

```diff
@@ -2496,12 +2496,29 @@ def __getstate__(self) -> dict:
         ):
             state["local_replay_buffer"] = self.local_replay_buffer.get_state()
 
+        # Save state of our counters.
+        state["counters"] = self._counters.copy()
+
         if self.train_exec_impl is not None:
             state["train_exec_impl"] = self.train_exec_impl.shared_metrics.get().save()
 
         return state
 
     def __setstate__(self, state: dict):
+        # Restore counters.
+        global_vars = {}
+        if "counters" in state:
+            # Don't recreate `self._counters` (it's already a defaultdict),
+            # just set its values.
+            for key, value in state["counters"].items():
+                self._counters[key] = value
+
+            # Set global vars.
+            if hasattr(self, "workers"):
+                global_vars = {
```

Comment:
maybe i'm not seeing it here, but where does global_vars get used?

Reply:
actually more specifically, I don't know why we should construct a global_vars section ourselves ...